### PR TITLE
Add edit page titles for references

### PR DESCRIPTION
**Pull request by JonF1**
## Context
There were no edit page titles for adding references. This meant there were duplicate page titles. This is not good for screen reader users and would be picked up on an accessibility review.

## Changes proposed in this pull request
* Add edit page titles for reference flow on candidate side

## Guidance to review
Add references and then edit to verify unique page titles

## Link to Trello card

https://trello.com/c/jhwFOJEX/514-references-bug-party-unique-page-titles

## Things to check

- [ ] If the code removes any existing feature flags, a data migration has also been added to delete the entry from the database
- [ ] This code does not rely on migrations in the same Pull Request
- [ ] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [ ] If this code adds a column to the DB, decide whether it needs to be in analytics yml file or analytics blocklist
- [ ] API release notes have been updated if necessary
- [ ] Required environment variables have been updated [added to the Azure KeyVault](/docs/environment-variables.md#deploy-pipeline)

### CODE REVIEW

File: config/locales/en.yml [hunk 1]
```diff
@@ -185,9 +185,13 @@ en:
     new_references: References
     new_references_start: Choose a referee
     new_references_name: What is the referee’s name?
+    new_references_edit_name: Edit the name of the referee
```
Comment (steventux) [inline config/locales/en.yml:188]: Not part of this PR but these keys are not really leveraging YAML nesting very well.

**Review by steventux — APPROVED** [review]

File: config/locales/en.yml (hunk 1 above)
Comment (JonF1) [reply config/locales/en.yml:188]: I'll card up
